### ISSUE TITLE
Question: what modules are optional?

### ISSUE THREAD

**Issue by punksta** [unimodules/react-native-unimodules#95]
Hello,

I am adding https://github.com/expo/expo/tree/master/packages/expo-apple-authentication into react-native (non-expo) project and would like to avoid adding unnesesarry ui-modules.

It would be helpful if you document which ui-modules are required for minimal set up and which are optional.

Thanks

**Comment by sjchmiela** [comment]
Hey @punksta!

First of all, let me remove some small misconception that may have arisen — unimodules may, but not have anything to do with the UI, so not every unimodule is a UI module. 😅 

When it comes to your question, we consider `react-native-unimodules` to provide a foundation, a base set of unimodules that should be helpful when integrating more dependencies. For `expo-apple-authentication`, as its `package.json` would suggest, the required modules are only [`@unimodules/core` and `@unimodules/react-native-adapter`](https://github.com/expo/expo/blob/ceee34ed8b8b151330ee0f4c1388f3ebf3b6db9b/packages/expo-apple-authentication/package.json#L34-L35). The adapter in turn depends on [`unimodules-font-interface`, `unimodules-image-loader-interface` and `unimodules-permissions-interface`](https://github.com/expo/expo/blob/ceee34ed8b8b151330ee0f4c1388f3ebf3b6db9b/packages/%40unimodules/react-native-adapter/package.json#L40-L42).

So, to sum up, if you feel you need to strip the setup down as much as possible, you could get away with:

- `@unimodules/core`
- `@unimodules/react-native-adapter`
- `unimodules-font-interface`
- `unimodules-image-loader-interface`
- `unimodules-permissions-interface`

Note that interface modules are… just interfaces, so their impact on the compilation time or size is minimal, while core and adapter are just what you need for `expo-apple-authentication` to work properly.

**Comment by punksta** [comment]
@sjchmiela thank you for such a detailed answer! I will try your suggestion this week. 

I was thinking not only about compilation time but about adding unnecessary permissions to my app.

**Comment by nolife08021** [comment]
@punksta Hi, did you able to add  https://github.com/expo/expo/tree/master/packages/expo-apple-authentication in your app?

I'm facing some errors when running **pod install** by using react-native-unimodules. If you successfully add and use the expo-apple-authentication in your app, can you give me some advice/guide about how to install it?
Thanks

My Error when running **pod install**
```
Analyzing dependencies
[!] Unable to find a specification for `React-Core` depended upon by `UMReactNativeAdapter`

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

**Comment by sjchmiela** [comment]
Hey @nolife08021, this issue is not related to your issue, I am replying now to your original https://github.com/unimodules/react-native-unimodules/issues/100. 🙂

**Comment by punksta** [comment]
@sjchmiela edited my Podfile according your suggestion. This worked:

```
use_unimodules!(exclude: [
        'expo-face-detector',
        'expo-constants',
        'expo-file-system',
        'expo-permissions',
        'expo-app-loader-provider',
        'unimodules-barcode-scanner-interface',
        'unimodules-camera-interface',
        'unimodules-constants-interface',
        'unimodules-face-detector-interface',
        'unimodules-file-system-interface',
        'unimodules-sensors-interface',
        'unimodules-task-manager-interface'
])
```

thanks
